Lib/javascript/jsc/javascriptstrings.swg

```diff
@@ -8,34 +8,32 @@ SWIG_JSC_AsCharPtrAndSize(JSContextRef context, JSValueRef valRef, char** cptr,
   if(JSValueIsString(context, valRef)) {
     JSStringRef js_str = JSValueToStringCopy(context, valRef, NULL);
     size_t len = JSStringGetMaximumUTF8CStringSize(js_str);
-    char* cstr = (char*) %new_array(len, char);
-    /* JSStringGetUTF8CString returns the length including 0-terminator */
-    len = JSStringGetUTF8CString(js_str, cstr, len);
-
-    if(alloc) *alloc = SWIG_NEWOBJ;
+    if(cptr) {
```

Comment:
This seems unrelated to the "isolate-aware" changes unless I'm missing something.  This is quite a large patch so including unrelated changes (even if they are useful changes) is not helpful for review.

Reply:
https://github.com/swig/swig/pull/1973#issuecomment-830243478 and https://github.com/swig/swig/pull/1973#issuecomment-846629495. And please be specific.

Reply:
From the linked comments:
> If it's considered too messy, then please state so. If it would be appropriate to break it down to separate requests, the please state so.

OK, I think this part would be better broken out as a separate change as it doesn't seem to be related to "isolate-aware init".  I would struggle to be more specific as I lack information on what's intended, as all I seem to have to go on is the (rather non-specific!) commit message:

```
Lib/javascript/*/javascriptstrings.swg: fix logical errors.

And improve readability.
````

Reply:
Removed the commit in question.